starfish/core/spots/DecodeSpots/test/test_composite_codebook.py

```diff
@@ -0,0 +1,340 @@
+        codebook=codebook
+    )
+    sequential_stack = sequential_stack.sel(indexers={Axes.ROUND: (3, 6)})
+
+    # create codebook with combined target values
+    combined = [
+        {
+            Features.CODEWORD: [
+                {Axes.ROUND.value: 0, Axes.CH.value: 0, Features.CODE_VALUE: 1},
+                {Axes.ROUND.value: 2, Axes.CH.value: 1, Features.CODE_VALUE: 1}
+            ],
+            Features.TARGET: "GENE_A"
+        },
+        {
+            Features.CODEWORD: [
+                {Axes.ROUND.value: 0, Axes.CH.value: 1, Features.CODE_VALUE: 1},
+                {Axes.ROUND.value: 1, Axes.CH.value: 2, Features.CODE_VALUE: 1}
+            ],
+            Features.TARGET: "GENE_B"
+        },
+        {
+            Features.CODEWORD: [
+                {Axes.ROUND.value: 3, Axes.CH.value: 1, Features.CODE_VALUE: 1}
+            ],
+            Features.TARGET: "GENE_C"
+        },
+        {
+            Features.CODEWORD: [
+                {Axes.ROUND.value: 4, Axes.CH.value: 0, Features.CODE_VALUE: 1}
+            ],
+            Features.TARGET: "GENE_D"
+        },
+        {
+            Features.CODEWORD: [
+                {Axes.ROUND.value: 5, Axes.CH.value: 1, Features.CODE_VALUE: 1}
+            ],
+            Features.TARGET: "GENE_E"
+        }
+
+    ]
+    codebook = Codebook.from_code_array(combined)
+    return codebook, multiplexed_stack, sequential_stack
+
+
+gaussian_spot_detector = simple_gaussian_spot_detector()
+
+
+def test_composite_codebook_decoding():
+    # create an ImageStack where rounds 0,1 contain multiplexed data and round 2 contains
+    # sequential data
+    codebook, stack = composite_codebook()
+
+    # select out the different portions of the imagestack
+    multiplexed_stack = stack.sel(indexers={Axes.ROUND: (0, 1)})
+    sequential_stack = stack.sel(indexers={Axes.ROUND: 2})
+
+    # select out the corresponding portions of the codebooks
+    codebook_multiplexed = codebook.get_partial(indexers={Axes.ROUND: (0, 1)})
+    codebook_sequential = codebook.get_partial(indexers={Axes.ROUND: 2})
+
+    # find spots
+    reference_image = multiplexed_stack.reduce((Axes.ROUND, Axes.CH), func="max")
+    multiplexed_spots = gaussian_spot_detector.run(
+        image_stack=multiplexed_stack, reference_image=reference_image)
+
+    sequential_spots = gaussian_spot_detector.run(image_stack=sequential_stack)
+
+    # decode
+    per_round_decoder = PerRoundMaxChannel(codebook=codebook_multiplexed)
+    decoded_multiplexed = per_round_decoder.run(spots=multiplexed_spots)
+    assert np.array_equal(decoded_multiplexed[Features.TARGET].values, ['GENE_B', 'GENE_A'])
+
+    simple_lookup_decoder = SimpleLookupDecoder(codebook=codebook_sequential)
+    decoded_sequential = simple_lookup_decoder.run(spots=sequential_spots)
+    assert np.array_equal(decoded_sequential[Features.TARGET].values, ['GENE_C', 'GENE_D'])
+
+
+def test_composite_codebook_decoding_seperate_sized_stacks():
```

Comment:
It would be good if this had a short description like `test_composite_codebook_decoding`.